### DISCUSSION TITLE
Restore library without losing data?

### DISCUSSION THREAD

**Discussion by anon767346** [stashapp/stash#4352] [Q&A] [answered]
I had to delete the containers with `docker compose down` and after bringing them up again all the scene, performers and tags are gone.

How do I restore the data from disk? Also how to prevent this from happening again when recreating the container?

**Comment by Asuryan34** [comment]
Hello.
Good question. Same things happened to me. I was OK because I just transferred my library from my old NAS  to the new so id didn't lone anything but ... As I know, you can't recover it since it was just wiped (without going to the trash (where it could have been recovered)

my idea was to host the file in an external folder. It's possible on a windows installation but I didn't find the way on an installation under docker

**Comment by DogmaDragon** [comment, marked as answer]
> Docker throws away everything when you create a new container. So you use mount points to store things outside the container. Use a mount point   /root/.stash to store the database etc. if you use docker-compose it should have this set up with folder mounts

Message from a helpful person on Discord
